pharmacy-api/src/Nuvem.PharmacyManagement.PharmacyServices/PharmacyService.cs

```diff
@@ -1,96 +1,107 @@
+            if (pharmacy is not null)
+            {
+                List<Pharmacy> pharmacyList = new List<Pharmacy> { pharmacy };
+                return pharmacyList;
+            }
+            else
+            {
+                return null;
+            }
+        }
+        return await Task.FromResult(_dbContext.Pharmacy.ToList());
     }
 
     public async Task<Pharmacy?> UpdatePharmacyAsync(Pharmacy pharmacy)
     {
         var existingPharmacy = await GetPharmacyByIdAsync(pharmacy.PharmacyId);
-            if (existingPharmacy is null) 
-            {
-                _logger.LogInformation("Pharmacy to update not found!");
-                return null;
-            }
-            pharmacy.UpdatedDate = DateTime.Now;
+        if (existingPharmacy is null)
+        {
+            _logger.LogInformation("Pharmacy to update not found!");
+            return null;
+        }
+        pharmacy.UpdatedDate = DateTime.Now;
 
-            _dbContext.Pharmacy.Attach(existingPharmacy);
+        _dbContext.Pharmacy.Attach(existingPharmacy);
 
-            var entry = _dbContext.Instance.Entry(existingPharmacy);
-            entry.CurrentValues.SetValues(pharmacy);
+        var entry = _dbContext.Instance.Entry(existingPharmacy);
+        entry.CurrentValues.SetValues(pharmacy);
 
-            entry.Property("PharmacyId").IsModified = false;
-            entry.Property("CreatedDate").IsModified = false;   
+        entry.Property("PharmacyId").IsModified = false;
+        entry.Property("CreatedDate").IsModified = false;
 
-            await _dbContext.SaveChangesAsync();  
-            return existingPharmacy; 
-    }   
+        await _dbContext.SaveChangesAsync();
+        return existingPharmacy;
+    }
 
     public async Task<PharmacyDisplayResult<Pharmacy>?> PaginatedPharmacyListAsync(ParameterCollection parameters)
     {
-        if(parameters.PageSize == 0) parameters.PageSize = 5;
+        if (parameters.PageSize == 0) parameters.PageSize = 5;
         PharmacyDisplayResult<Pharmacy> result = new();
-        List<Pharmacy> pharmacyList = await GetPharmacieListAsync();
+        List<Pharmacy>? pharmacyList = await GetPharmacieListAsync();
+        if (pharmacyList is null)
+        {
+            return null;
+        }
         result.TotalCount = pharmacyList.Count;
         int skip = parameters.Page * parameters.PageSize;
         var canPage = skip < pharmacyList.Count;
-        if(!canPage) return null;
-        result.List = pharmacyList.Select(p=> p)
+        if (!canPage) return null;
+        result.List = pharmacyList.Select(p => p)
                         .Skip(skip)
                         .Take(parameters.PageSize)
                         .ToList();
 
         return result;
     }
+
+    private async Task<Pharmacy?> GetPharmacyByIdAsync(int id)
+    {
+        return await _dbContext.Pharmacy.FirstOrDefaultAsync(x => x.PharmacyId == id);
```

Comment:
It can be used interchangeably in this case. 
Here is the difference though:

FindAsync():
accepts primary key
optimized for finding an entity with a primary key, by returning the entity without hitting the database if the entity was tracked.
can't use the include method with findasync.
returns default value if no match was found.


FirstOrDefaultAsync() :
accepts expression.
better choice over FindAsync if related entities are needed.
returns default value if no match was found.